### ISSUE TITLE
SDDM should handle situations where using OpenGL is not working or feasible to use

### ISSUE THREAD

**Issue by karolherbst** [sddm/sddm#1095]
there are some issues regarding broken system setups or old OpenGL max versions where sddm doesn't work because no OpenGL context could be created. (#799 #575 #376)

But it is crucial that a login manager works no matter what the state of the graphics driver setup is as something stupid can always happen, like efifb is the only fb device you get and X has to use the fbdev DDX where OpenGL is not even possible if your system gl is nvidia or some other implementation which doesn't provide a software fallback.

It would be nice to have some proper fallback in such cases, so that you can still do a login.

**Comment by davidedmundson** [comment]
Yeah, we started doing that in Plasma, on event of a crash we restart with QT_QUICK_BACKEND=software

probably worth doing here.

**Comment by plfiorini** [comment]
The software backend might not work if the theme uses shaders.
If we can detect it we might load the embedded theme and be even more reliable.

**Comment by davidedmundson** [comment]
if you do use shaders with software mode all that happens is they don't render.  So mostly it's just a loss of shadows. 

However, we should probably go to the default theme in the event of a crash too.

**Comment by plfiorini** [comment]
We already do that :)
